AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/ReconcileAndLocalSave/RemoteSyncReconciler.swift

```diff
@@ -21,61 +21,25 @@ struct RemoteSyncReconciler {
     }
 
     static func reconcile(remoteModel: RemoteModel,
-                          to localModel: LocalModel?,
-                          storageAdapter: StorageEngineAdapter) -> Disposition {
-
-        let pendingMutations: [MutationEvent]
-        switch getPendingMutations(forModelId: remoteModel.model.id, storageAdapter: storageAdapter) {
-        case .failure(let dataStoreError):
-            return .error(dataStoreError)
-        case .success(let mutationEvents):
-            pendingMutations = mutationEvents
-        }
-
-        return disposition(for: remoteModel,
-                           localModel: localModel,
-                           pendingMutations: pendingMutations)
-    }
-
-    private static func getPendingMutations(forModelId modelId: Model.Identifier,
-                                            storageAdapter: StorageEngineAdapter) -> DataStoreResult<[MutationEvent]> {
-        let semaphore = DispatchSemaphore(value: 1)
-        var pendingMutationResultFromQuery: DataStoreResult<[MutationEvent]>?
-        MutationEvent.pendingMutationEvents(forModelId: modelId,
-                                            storageAdapter: storageAdapter) {
-                                                pendingMutationResultFromQuery = $0
-                                                semaphore.signal()
-        }
-        semaphore.wait()
-
-        guard let pendingMutationResult = pendingMutationResultFromQuery else {
-            let dataStoreError = DataStoreError.unknown("Unable to query pending mutation events",
-                                                        AmplifyErrorMessages.shouldNotHappenReportBugToAWS())
-            return .failure(dataStoreError)
-        }
-
-        return pendingMutationResult
-    }
-
-    private static func disposition(for remoteModel: RemoteModel,
-                                    localModel: LocalModel?,
-                                    pendingMutations: [MutationEvent]) -> Disposition {
+                          to localMetadata: LocalMetadata?,
+                          pendingMutations: [MutationEvent]) -> Disposition {
 
         guard pendingMutations.isEmpty else {
             return .dropRemoteModel
         }
 
-        guard let localModel = localModel else {
+        guard let localMetadata = localMetadata else {
             return .applyRemoteModel(remoteModel)
         }
 
         // Technically, we should never receive a subscription for a version we already have, but we'll be defensive
         // and make this check include the current version
-        if remoteModel.syncMetadata.version >= localModel.syncMetadata.version {
+        if remoteModel.syncMetadata.version >= localMetadata.version {
```

Comment:
Seems like before, we were querying the local model, and looking at it's attached metadata.. now it looks like we are querying the local metadata directly.
1.  Why did we make this change?
2.  Is there any difference between the two versions of the meta data? (in terms of timing -- are there any out of sync errors that can happen?)

Reply:
1. The decisions we make on conflict resolution are based solely on the presence or not of a pending mutation; and a comparison of the versions between remote and local sync metadata.
2. We only take sync metadata from the service, so they should always be in sync.